### DISCUSSION TITLE
How to change the publication rate of /joint_states topic ?

### DISCUSSION THREAD

**Discussion by MohamedOmar2014** [ros-industrial/universal_robot#655] [Q&A] [answered]
I'm working on UR10 Robot, I'm subscribing to /joint_stets topic to get the sensor data (position, velocity), I'm trying to increase the publication rate (default is 125 Hz) of the topic in order to smooth the resulted position and velocity curves. please, can you tell me, how can I do it?

**Comment by gavanderhoorn** [comment, marked as answer]
If this is a real robot: you can't.

`JointState` messages are published at the rate the UR control box provides them (ie: sends out robot state).

There would be nothing to publish at a faster rate.

(e-Series controllers run at 500 Hz, so buying a new UR could be a work-around)

**Reply by MohamedOmar2014** [reply]
Yes, I'm working on a real one, so, it seems that it is impossible
